### PR TITLE
patch/erc20

### PR DESCRIPTION
**Pull request by dudong2**
## Description

- deprecate `RegisterCoinProposal`, `RegisterERC20Proposal` and `ToggleTokenConversionProposal` that is used for gov tx submit-proposal-legacy
- add tx command for `NewRegisterCoinProposalCmd`, `NewRegisterERC20ProposalCmd` and `NewToggleTokenConversionProposalCmd`
- replace naming of legacy amino codec

Closes: #XXXX

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] targeted the correct branch (see [PR Targeting](https://github.com/canto-network/canto/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] included the necessary unit and integration tests
- [ ] reviewed "Files changed" and left comments if necessary
 
### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

**Review by poorphd — COMMENTED** [review]
LGTM, however,  when submitting a proposal with the `erc20` subcommand, there is an issue where the deposit needs to be made in `stake` instead of `acanto`. It seems necessary to resolve this issue separately.

**Comment by dudong2** [comment]
> LGTM, however, when submitting a proposal with the `erc20` subcommand, there is an issue where the deposit needs to be made in `stake` instead of `acanto`. It seems necessary to resolve this issue separately.

It was resolved at [govshuttl pr](https://github.com/b-harvest/Canto/pull/60/files#diff-408f2ee929058c9e991c5af8d003eaded2c24e8616b94e48de84031a80397c0c)

File: x/erc20/types/codec.go [hunk 1]
```diff
@@ -63,10 +56,10 @@ func RegisterInterfaces(registry codectypes.InterfaceRegistry) {
...
-	cdc.RegisterConcrete(&MsgRegisterCoin{}, "canto/RegisterCoinProposal", nil)
-	cdc.RegisterConcrete(&MsgRegisterERC20{}, "canto/RegisterERC20Proposal", nil)
-	cdc.RegisterConcrete(&MsgConvertERC20{}, convertERC20Name, nil)
-	cdc.RegisterConcrete(&MsgConvertCoin{}, convertCoinName, nil)
-	cdc.RegisterConcrete(&MsgUpdateParams{}, msgUpdateParams, nil)
-	cdc.RegisterConcrete(&Params{}, "canto/Params", nil)
+	cdc.RegisterConcrete(&MsgRegisterCoin{}, "canto/MsgRegisterCoin", nil)
+	cdc.RegisterConcrete(&MsgRegisterERC20{}, "canto/MsgRegisterERC20", nil)
+	cdc.RegisterConcrete(&MsgConvertCoin{}, "canto/MsgConvertCoin", nil)
+	cdc.RegisterConcrete(&MsgConvertERC20{}, "canto/MsgConvertERC20", nil)
```
Comment (zsystm) [inline x/erc20/types/codec.go:62]: @dudong2 
- Why don't we use `canto/x/erc20/MsgRegisterCoin` and `canto/x/erc20/MsgRegisterERC20` here?
- I know it follows `amino.name` in `tx.proto`, but curious for naming. I think you were trying to unify the naming like `canto/x/<module>/type`

**Review by zsystm — COMMENTED** [review]
@dudong2 
LGTM except one small comment for naming

File: x/erc20/types/codec.go (hunk 1 above)
Comment (dudong2) [reply x/erc20/types/codec.go:62]: good opinion. but we need to consider consistency not only with canto but also with cosmos-sdk.
in the sdk, it follows the form `cosmos-sdk/Msg...`, so Canto should also adopt this format.

Comment (zsystm) [reply x/erc20/types/codec.go:62]: @dudong2 thanks for your details :)

now LGTM

**Review by zsystm — APPROVED** [review]

**Review by poorphd — APPROVED** [review]

**Pull request merged**